hartshorn-core/src/main/java/org/dockbox/hartshorn/core/context/element/TypeContext.java

```diff
@@ -380,6 +380,16 @@ public boolean isNative() {
         return this.isNative;
     }
 
+    public boolean isArray() {
+        return this.type().isArray();
+    }
+
+    public Exceptional<TypeContext<?>> elementType() {
+        return this.isArray()
+                ? Exceptional.of(of(this.type().getComponentType()))
```

Comment:
Preferably this should be cached, to avoid having to repeat validation on TypeContext and Exceptional each time this method is called. The same is done for e.g. the default constructor.
https://github.com/GuusLieben/Hartshorn/blob/a48c2eafe4240e3bd91713ca952b1ae9bc95defc/hartshorn-core/src/main/java/org/dockbox/hartshorn/core/context/element/TypeContext.java#L121